docs/advanced/customization-guide.md

```diff
@@ -16,8 +15,1030 @@ published: false
 
 ---
 
-***WORK IN PROGRESS.***
+## Overview
+
+NFD provides multiple extension points for vendor and application specific
+labeling:
+
+- [NodeFeatureRule](#nodefeaturerule-custom-resource) objects provide a way to
+  deploy custom labeling rules via the Kubernetes API
+- [*local* feature source](#local-feature-source) of nfd-worker creates
+  labels by executing hooks and reading files
+- [*custom* feature source](#custom-feature-source) of nfd-worker creates
+  labels based on user-specified rules
+
+## NodeFeatureRule custom resource
+
+NodeFeatureRule objects provide an easy way to create vendor or application
+specific labels. It uses a flexible rule-based mechanism for creating labels
+based on node feature.
+
+### A NodeFeatureRule example
+
+Consider the following referential example:
+
+```yaml
+apiVersion: nfd.k8s-sigs.io/v1alpha1
+kind: NodeFeatureRule
+metadata:
+  name: my-sample-rule-object
+spec:
+  rules:
+    - name: "my sample rule"
+      labels:
+        "my-sample-feature": "true"
+      matchFeatures:
+        - feature: kernel.loadedmodule
+          matchExpressions:
+            dummy: {op: Exists}
+        - feature: kernel.config
+          matchExpressions:
+            X86: {op: In, value: ["y"]}
+```
+
+It specifies one rule which creates node label
+`feature.node.kubenernetes.io/my-sample-feature=true` if both of the following
+conditions are true (`matchFeatures` implements a logical AND over the
+matchers):
+
+- The `dummy` network driver module has been loaded
+- X86 option in kernel config is set to `=y`
+
+Create a NodeFeatureRule with a yaml file:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/node-feature-discovery/{{ site.release }}/examples/nodefeaturerule.yaml
+```
+
+Now, on X86 platforms the feature label appears after doing `modprobe dummy` on
+a system and correspondingly the label is removed after `rmmod dummy`. Note a
+re-labeling delay up to the sleep-interval of nfd-worker (1 minute by default).
+
+### NodeFeatureRule controller
+
+NFD-Master acts as the controller for NodeFeatureRule objects. It applies these
```

Comment:
```suggestion
NFD-Master acts as the controller for `NodeFeatureRule` objects. It applies these
```

ditto for other cases too

Reply:
Yeah, maybe it makes it more readable. Changed as suggested. I also changed *local* and *custom* similarly